ecommerce/extensions/fulfillment/modules.py

```diff
@@ -116,17 +120,15 @@ def execute_external_enrollment(self, order, supported_lines):
                 "course_mode": mode_for_product(line.product),
                 "course_id": line.product.attr.course_key
             }
-
-            url = get_lms_url('/openedx_external_enrollments/api/v0/external-enrollment')
-            headers = {"Authorization": "token", "Accept":"application/json", "Content-Type": "application/json"}
+            api = EdxRestApiClient(
+                url=get_external_enrollment_api_url(),
+                jwt=order.site.siteconfiguration.access_token,
+                append_slash=False,
+            )
+            logging.info("Calling enrollment plugin with: %s", payload)
             try:
-                logging.info('calling enrollment plugin with: %s', payload)
-                response = requests.post(url, headers=headers, data=json.dumps(payload))
-                if response.ok:
-                    data = response.json()
-                    logging.info("External enrollment call completed - data %s", data)
-                else:
-                    logging.error("Error calling plugin: %s", response.json())
+                data = getattr(api, "external-enrollment").post(payload)
```

Comment:
An attribute error could be raised here, please add a default value. @andrey-canon

Reply:
There is no default value, the api object works in the following way, if you write api.whatever_you_want.post()  this will make a post request to  http://api_root/whatever_you_want, in this case will be http://api_root/external-enrollment, so the attribute error  will not be raise, because you can write all that you want, and the request will use that path

Reply:
If so, why the getattr? I can coding as: api.external-enrollment.post(payload)

Reply:
the dash is an arithmetic operator, that is similar to write  a.b **-** c.post()

Reply:
All right, that is the path, got it. @andrey-canon